src/Transport/KubectlTransport.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Consolidation\SiteProcess\Transport;
+
+use Consolidation\SiteProcess\SiteProcess;
+use Consolidation\SiteProcess\Util\Escape;
+use Consolidation\SiteAlias\SiteAliasInterface;
+use Consolidation\SiteProcess\Util\Shell;
+
+/**
+ * KubectlTransport knows how to wrap a command such that it runs in a container
+ * on Kubernetes via kubectl.
+ */
+class KubectlTransport implements TransportInterface
+{
+    /** @var bool */
+    protected $tty;
+
+    /** @var \Consolidation\SiteAlias\SiteAliasInterface */
+    protected $siteAlias;
+
+    public function __construct(SiteAliasInterface $siteAlias)
+    {
+        $this->siteAlias = $siteAlias;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function configure(SiteProcess $process)
+    {
+        $this->tty = $process->isTty();
+    }
+
+    /**
+     * inheritdoc
+     */
+    public function wrap($args)
+    {
+        # TODO: How/where do we complain if a required argument is not available?
```

Comment:
@greg-1-anderson How/where do we complain if a required argument in is not available in the site alias?  Is there a "standard" way (e.g. a specific custom exception that should be thrown) in this repo?

Reply:
Throwing an exception would be best; however, at the moment, this library does not provide any custom exception types. You could provide one, if you wanted, or maybe just use InvalidArgumentException. Maybe that's just similar enough to be confusing, though. Recommendations welcome.

Reply:
I would leave the TODO there, merge it, and make a new issue to address the need for custom exceptions. It needs to be designed well.